### PR TITLE
add indexing options to CreateCollectionOptions

### PR DESCRIPTION
**Pull request by vkarpov15**
**What this PR does**:

Adds `indexing` to `CreateCollectionOptions`. The `allow: never` and `deny: never` are [unfortunately necessary](https://stackoverflow.com/questions/37688318/typescript-interface-possible-to-make-one-or-the-other-properties-required).

I also fixed a couple of build failures in the test cases that I found while testing the `CreateCollectionOptions` type changes.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [ ] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: tests/collections/options.test.ts [hunk 1]
```diff
@@ -128,7 +128,7 @@ describe('Options tests', async () => {
             //rawResult options should be cleaned up by stargate-mongoose, but 'upsert' should be preserved
             const updateOneResp = await Product.updateOne({ name: 'Product 4' },
                 { $set : { isCertified : true }, $inc: { price: 5 } },
-                { upsert: true, rawResult: false, sort: { name : 1 } }
+                { upsert: true, rawResult: false, sort: { name : 1 } } as unknown as Record<string, never>
```
Comment (kathirsvn) [inline tests/collections/options.test.ts:135]: @vkarpov15  I'm sorry could you please explain why `as unknown as Record<string, never>` is required here now?

File: src/collections/options.ts [hunk 2]
```diff
@@ -123,8 +123,11 @@ export const updateOneInternalOptionsKeys: Set<string> = new Set(
     Object.keys(new _UpdateOneOptions)
 );
 
+type IndexingOptions = { deny: string[], allow: never } | { allow: string[], deny: never };
```
Comment (kathirsvn) [inline src/collections/options.ts]: Can we add a couple of tests that pass the `allow` and `deny` params please?

File: tests/collections/options.test.ts (hunk 1 above)
Comment (vkarpov15) [reply tests/collections/options.test.ts:135]: TypeScript complains that `rawResult` is not an allowed property. Mongoose 8.1 made the `options` parameter for `Product.updateOne()` a bit more strict: Mongoose now disallows any options for `Product.updateOne()` that aren't supported by `updateOne()`, including `rawResult` and `sort`.

File: src/collections/options.ts (hunk 2 above)
Comment (vkarpov15) [reply src/collections/options.ts]: Done in 6b01675. I needed to bump the JSON API and stargate versions in api-compatibility.versions.json though

**Comment by vkarpov15** [comment]
@kathirsvn this should be ready for review again. One remaining test failure should be fixed by release of https://github.com/stargate/jsonapi/pull/863. I ended up having to do a bunch of test refactor to avoid dropping and recreating collections. because that was making the full test suite take 35 minutes to run on GH actions against latest JSON API, and I was facing an issue that I could only repro by running the full test suite.

**Comment by kathirsvn** [comment]
@vkarpov15 It looks like, there is a conflict now on this branch. Sorry for the delay, I can review this today once the conflicts are resolved and we can try closing this soon, thanks!

**Comment by vkarpov15** [comment]
@kathirsvn I resolved all merge conflicts and fixed tests, can you please take a look?

**Review by kathirsvn — APPROVED** [review]
LGTM

**Pull request merged**